src/preprocessing/passes/symmetry_detect.cpp

```diff
@@ -0,0 +1,456 @@
+/*********************                                                        */
+/*! \file symmetry_detect.cpp
+ ** \verbatim
+ ** Top contributors (to current version):
+ **  Paul Meng, Andrew Reynolds
+ ** This file is part of the CVC4 project.
+ ** Copyright (c) 2009-2018 by the authors listed in the file AUTHORS
+ ** in the top-level source directory) and their institutional affiliations.
+ ** All rights reserved.  See the file COPYING in the top-level source
+ ** directory for licensing information.\endverbatim
+ **
+ ** \brief Symmetry detection module
+ **/
+
+#include "preprocessing/passes/symmetry_detect.h"
+#include "theory/quantifiers/term_util.h"
+#include "theory/rewriter.h"
+
+using namespace std;
+
+namespace CVC4 {
+namespace preprocessing {
+namespace passes {
+
+SymmetryDetect::Partition SymmetryDetect::detect(const vector<Node>& assertions)
+{
+  Partition p =
+      findPartitions(NodeManager::currentNM()->mkNode(kind::AND, assertions));
+  Trace("sym-dt") << endl
+                  << "------------------------------ The Final Partition "
+                     "------------------------------"
+                  << endl;
+  printPartition(p);
+  return p;
+}
+
+void SymmetryDetect::getPartition(vector<vector<Node> >& parts,
+                                  const vector<Node>& assertions)
+{
+  Partition p = detect(assertions);
+
+  for (map<Node, vector<Node> >::iterator subvar_to_vars_it =
```

Comment:
Use a `const_iterator`. Could also write these lines simply as: `for (const auto& subvar_vars : p.d_subvar_to_vars)` (this also applies to other for loops in the PR).